torch/csrc/jit/passes/quantization.cpp

```diff
@@ -1488,96 +1500,109 @@ graph(%self, %x):
-        conv_submodule.setattr("bias", std::get<1>(new_w_b));
-      } else {
-        // conv module has an existing non-Tensor bias
-        if (conv_submodule.hasattr("bias")) {
-          if (conv_submodule._ivalue()->type()->hasConstant("bias")) {
-            GRAPH_UPDATE("Removing bias constant from conv module");
-            conv_submodule.type()->unsafeRemoveConstant("bias");
-          } else {
-            TORCH_CHECK(
-                conv_submodule.type()->findAttributeSlot("bias").has_value());
-            GRAPH_UPDATE(
-                "Removing existing non-Tensor bias attribute from conv module");
-            conv_submodule._ivalue()->unsafeRemoveAttr("bias");
-            conv_submodule.type()->unsafeRemoveAttribute("bias");
-          }
+        if (is_folded_graph) {
+          GRAPH_DEBUG("ConvBn in the graph is already folded, skipping modifications on"
+                      " graph and type");
+          continue;
+        }
+        // We are using a separate vector for saving Values we want to rewrite to
+        // make sure that the order in which we perform these transformations is
+        // deterministic. Iterating through keys of rewrite_map would result in
+        // non-determinism that might not manifest as a bug now, but can bite us
+        // later.
+        values_to_rewrite_.push_back(matched_bn->output());
+        rewrite_map_[matched_bn->output()] = matched_conv->output();
+        GRAPH_UPDATE(
+            "Rewriting %",
+            matched_bn->output()->debugName(),
+            " with %",
+            matched_conv->output()->debugName());
+
+        nodes_to_delete_.insert(matched_bn);
+        GRAPH_UPDATE("Deleting ", *matched_bn);
+
+        auto slot = conv_submodule.type()->getAttributeSlot("bias");
+        // We need to remove the attribute field before adding parameter field
+        if (!conv_submodule.type()->is_parameter(slot)) {
+          GRAPH_UPDATE(
+              "Removing existing bias attribute from conv module");
+          conv_submodule._ivalue()->unsafeRemoveAttr("bias");
+          conv_submodule.type()->unsafeRemoveAttribute("bias");
         }
-        conv_submodule.register_parameter("bias", std::get<1>(new_w_b), false);
-      }
-    }
 
-    // Perform planned rewritings
-    for (auto v : values_to_rewrite) {
-      v->replaceAllUsesWith(rewrite_map.at(v));
-    }
+      } // matches
+    } // methods
+  } // while
+}
 
-    // Perform planned deletions
-    for (auto n : nodes_to_delete) {
-      n->removeAllInputs();
-    }
-    for (auto n : nodes_to_delete) {
-      n->destroy();
-    }
+void FoldConvBatchNorm2dHelper::transform() {
+  for (const auto& item : conv_module_and_params_) {
+    script::Module conv(item.first);
+    auto w_b = item.second;
+    conv.setattr("weight", std::get<0>(w_b));
+    // registering a parameter field (not attribute field)
+    conv.register_parameter("bias", std::get<1>(w_b), false);
```

Comment:
What happens if `bias` already exists as a parameter?

Reply:
That's fine, we'll overwrite it

Reply:
this does two things, if parameter bias already exist, we'll overwrite, if not(we deleted bias attribute in analyze step), we'll register a bias parameter